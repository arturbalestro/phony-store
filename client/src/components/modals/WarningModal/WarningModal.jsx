import React, { useState, useEffect } from "react";
import { connect } from "react-redux";
import { Modal, ModalHeader, ModalBody, ModalFooter } from "reactstrap";
import { StyledButton } from "../../../styles/StyledButton";

export const WarningModal = ({ warningModal }) => {
  const [modal, setWarningModal] = useState(false);

  const modalClose = () => {
    setWarningModal(false);
  };

  useEffect(() => {
    if (warningModal) {
      setWarningModal(warningModal.modalOpen);
    }
  }, [warningModal]);

  if (warningModal) {
    return (
      <div>
        <Modal isOpen={modal} onClick={modalClose}>
          <ModalHeader>{warningModal.actionType}</ModalHeader>
          <ModalBody>{warningModal.modalMessage}</ModalBody>
          <ModalFooter>
            <StyledButton onClick={modalClose}>Close</StyledButton>
          </ModalFooter>
        </Modal>
      </div>
    );
  }

  return (
    <div>
      <Modal isOpen={modal} onClick={modalClose}>
        <ModalHeader>Error</ModalHeader>
        <ModalBody>An error has occurred</ModalBody>
        <ModalFooter>
          <StyledButton onClick={modalClose}>Close</StyledButton>
        </ModalFooter>
      </Modal>
    </div>
  );
};

//Mapping warningModal state
const mapStateToProps = (state) => ({ warningModal: state.warningModal });

export default connect(mapStateToProps, null)(WarningModal);
