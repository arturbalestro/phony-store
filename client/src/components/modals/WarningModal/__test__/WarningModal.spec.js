// @flow
import WarningModal from "../WarningModal";
import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

const middlewares = [];
const mockStore = configureStore(middlewares);
const initialState = {};
const store = mockStore(initialState);

test("<WarningModal /> renders correctly", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <WarningModal />
      </Provider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
