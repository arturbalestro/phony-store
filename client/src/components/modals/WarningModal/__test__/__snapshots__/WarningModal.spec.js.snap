// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WarningModal /> renders correctly 1`] = `<div />`;
